platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheServices.kt

```diff
@@ -82,6 +83,7 @@ class ConfigurationCacheServices : AbstractPluginServiceRegistry() {
         registration.run {
             add(ConfigurationCacheHost::class.java)
             add(ConfigurationCacheIO::class.java)
+            addProvider(IsolatedProjectEvaluationListenerProvider)
```

Comment:
💅  Why not simply `add(DefaultIsolatedProjectEvaluationListenerProvider::class.java)` ?

The best practice now would actually be
```
add(IsolatedProjectEvaluationListenerProvider::class.java, DefaultIsolatedProjectEvaluationListenerProvider::class.java)
```

in line with the upcoming strict scoping for all services.

In Kotlin, such a call also begs for an inline-reified helper 😃